tests/stdlib/tgetprotobyname.nim

```diff
@@ -1,9 +1,5 @@
 import nativesockets
 
-when not defined(netbsd):
-  # Ref: https://github.com/nim-lang/Nim/issues/15452 - NetBSD doesn't define an `ip` protocol
-  doAssert getProtoByName("ip") == 0
-
```

Comment:
Why is this removed at all?

Reply:
This test fails at my manjaro machine, and it my Arch linux machine. Aparently it also causes issues at NetBSD and who knows where else. Maybe it is better to not test for the wrapper that the backend actually supports an API? After all getProtoByName is just a wrapper, not some actual implementation.

Reply:
Yes I can do it. But I also want you to know that it really costs a lot of energy from me to manage many different pull requests that have dependencies on each other.

Reply:
What do you think about making a PR that only updates the tests, but all of them that are required for this PR to work? The tests should have been updated or even partially deleted long time ago. They don't specifically test the `callsite()` feature, they just use it for historic reasons, nothing else. They don't represent current Nim anymore at all.

Reply:
It sounds fine, though I think leaving them in here is a bit better for people reading this in the future, as this change have an effect on `callsite()`'s output and you get to figure out why was `callsite` purged from tests and what changed all at once.

I'd say that only this one test should be splitted out due to being completely irrelevant to this change.

Reply:
Mild preference for this PR. Just my two cents.